bump-tag

```diff
@@ -1,66 +1,268 @@
+    git_log="$(git log --pretty="format:%H${t}%G?" "${revision_range}" \
+                       --first-parent \
+                       | grep -v "${t}G$")"
+
 fi
-tagpfx=${tag:="${deftag}"}
-
-last_tag=$(git tag -l "${tagpfx}-*" | sort | tail -1)
-if [[ -n ${last_tag} ]]; then
-  echo "Verifying last tag ${last_tag}:"
-  (git tag -v "${last_tag}" | grep ^gpg:) || true
-  # again to not mask exit status of git with grep
-  git tag -v "${last_tag}" >/dev/null 2>&1
-  echo ""
-
-  echo "Differences between tag ${last_tag} and what you are about to sign:"
-  this_branch=$(git rev-parse --abbrev-ref HEAD)
-  env PAGER=cat git diff --color "${last_tag}..${this_branch}"
-
-  iter=1
-  ok=
-  while test -z "$ok"; do
-    this_tag=$(date "+${tagpfx}-%Y-%m-%d-v$(printf "%02d" ${iter})")
-    iter=$(( iter + 1))
-    case $( (
-      echo "${this_tag}"
-      echo "${last_tag}"
-    ) | sort | tail -1) in
-    "${last_tag}") ;;
-
-    "${this_tag}")
-      ok=yes
-      ;;
+
+if [[ ! -z "${git_log}" ]]; then
+    echo ""
+    echo -e "------WARNING: unsigned or untrusted commits after the last tag------"
+    echo "${git_log}"
+    echo -e "---------------------------------------------------------------------"
+    echo "Quick referens on how to configure signing of commits in ~/.gitconfig:"
+    echo "[user]"
+    echo "    signingkey = your-prefered-key-id"
+    echo "[commit]"
+    echo "    gpgsign = true"
+    echo ""
+    echo "EXITING the script. In order to tag your changes,"
+    echo "please make sure that you have configured signing of"
+    echo "your own commits and that the listed unsigned commits"
+    echo "have been made by a trusted party and are not malicous."
+    exit 1
+fi
+
+# Always check that the last commit of certain
+# sensitive files is trusted, without taking into
+# account whether the last tag was trusted or not.
+verify_last_commit "./scripts/jsonyaml-no-output.py"
+verify_last_commit "./bump-tag"
+
+# Test the syntax of each YAML-file to be tagged.
+for file in $(git diff --name-only "${last_tag}..${this_branch}" | egrep "^.*\.(yaml|yml)$"); do
+    if [[ -f "${file}" ]]; then
+        ./scripts/jsonyaml-no-output.py yaml "${file}"
+    fi
+done
+
+echo "Differences between tag ${last_tag} and what you are about to sign:"
+# With PAGER=cat, git diff will simply dump the output to the screen.
+# shellcheck disable=SC2037
+PAGER=cat git diff --color "${last_tag}..${this_branch}"
+
+# Iterate over the $last_tag until $this_tag is set to a later version
+iter=1
+ok=
+while [[ -z "${ok}" ]]; do
+    this_tag="$(date +"${tagpfx}-%Y-%m-%d-v$(printf "%02d" "${iter}")")"
+    iter="$(( iter + 1))"
+
+    case "$( (echo "${this_tag}"; echo "${last_tag}") | sort | tail -1 )" in
+        "${last_tag}")
+            ;;
+        "${this_tag}")
+            ok=yes
+            ;;
     esac
-  done
+done
 
-  if [ "${deftag}" != "${tagpfx}" ]; then
+if [ "${deftag}" != "${tagpfx}" ]; then
```

Comment:
We should use  double square brackets here, as is in the rest of the script
```
if [[ "${deftag}" != "${tagpfx}" ]]; then
```

Reply:
Agree :) I've therefore changed this line to be consistent with the rest of the script.